In [ ]:
import pandas as pd


from google.colab import files
uploaded = files.upload()

# To load the Customer file (replace 'customers.csv' with your actual filename)
# We use 'pd.read_csv' to open the file
df_customers = pd.read_csv('customers.csv')


print("Here is a peek at your customer data:")
display(df_customers.head())

Saving customers.csv to customers (1).csv
Here is a peek at your customer data:


,customer_id,signup_date,city_tier,age_group,acquisition_channel,loyalty_tier,preferred_category,skin_type,marketing_consent
0,CUST00001,2024-04-24,Tier 1,18-24,Instagram,Silver,Makeup,Normal,Yes
1,CUST00002,2025-06-01,Tier 2,25-34,Marketplace,Silver,Hair Care,Combination,Yes
2,CUST00003,2025-03-08,Tier 1,25-34,Influencer,NaN,Skin Care,Oily,Yes
3,CUST00004,2025-04-15,Tier 3,25-34,Google Search,NaN,Fragrance,Normal,No
4,CUST00005,2024-08-21,Tier 3,35-44,Organic,Gold,Hair Care,Combination,Yes


In [ ]:
import pandas as pd
from google.colab import files


uploaded = files.upload()


customers = pd.read_csv('customers.csv')
orders = pd.read_csv('orders.csv')
tickets = pd.read_csv('support_tickets.csv')

print("Missing values in Customer data:")
print(customers.isnull().sum())

print("\nMissing values in Order data:")
print(orders.isnull().sum())

Saving support_tickets.csv to support_tickets (1).csv
Missing values in Customer data:
customer_id               0
signup_date               0
city_tier                 0
age_group                 0
acquisition_channel       0
loyalty_tier           1386
preferred_category        0
skin_type               401
marketing_consent         0
dtype: int64

Missing values in Order data:
order_id          0
customer_id       0
order_date        0
category          0
quantity          0
gross_amount      0
discount_pct      0
delivery_days     0
returned          0
rating           80
dtype: int64


In [ ]:
import pandas as pd
from google.colab import files


uploaded = files.upload()


customers = pd.read_csv('customers.csv')
orders = pd.read_csv('orders.csv')
tickets = pd.read_csv('support_tickets.csv')


print("Missing values in Customer data:")
print(customers.isnull().sum())

print("\nMissing values in Order data:")
print(orders.isnull().sum())

Saving orders.csv to orders (2).csv
Missing values in Customer data:
customer_id               0
signup_date               0
city_tier                 0
age_group                 0
acquisition_channel       0
loyalty_tier           1386
preferred_category        0
skin_type               401
marketing_consent         0
dtype: int64

Missing values in Order data:
order_id          0
customer_id       0
order_date        0
category          0
quantity          0
gross_amount      0
discount_pct      0
delivery_days     0
returned          0
rating           80
dtype: int64


In [ ]:
import pandas as pd

# Now that all files are in the folder, just load them by name!
customers = pd.read_csv('customers.csv')
orders = pd.read_csv('orders.csv')
tickets = pd.read_csv('support_tickets.csv')

# Merge them: Add orders and tickets to your customer list

master_df = customers.merge(orders, on='customer_id', how='left')
master_df = master_df.merge(tickets, on='customer_id', how='left')


print("Total records in your Master Table:", len(master_df))
print("Missing values in your Master Table:")
print(master_df.isnull().sum())

Total records in your Master Table: 14850
Missing values in your Master Table:
customer_id               0
signup_date               0
city_tier                 0
age_group                 0
acquisition_channel       0
loyalty_tier           7975
preferred_category        0
skin_type              2434
marketing_consent         0
order_id                  0
order_date                0
category                  0
quantity                  0
gross_amount              0
discount_pct              0
delivery_days             0
returned                  0
rating                  116
ticket_id              3673
ticket_date            3673
issue_type             3673
support_channel        3673
resolution_hours       3673
sentiment_score        3673
reopened               3673
dtype: int64


In [ ]:
import pandas as pd
import os

# To check what the computer actually sees in the folder
print("Files in folder:", os.listdir())

# To load the file, but only if it's there
filename = 'customers.csv'

if filename in os.listdir():
    df = pd.read_csv(filename)
    print("Success! Here is your data:")
    display(df.head())
else:
    print(f"Error: I cannot find '{filename}' in the folder. Please drag and drop it into the folder icon on the left!")

Files in folder: ['.config', 'customers (1).csv', 'customers.csv', 'orders (1).csv', 'support_tickets.csv', 'orders (2).csv', 'support_tickets (1).csv', 'orders.csv', 'sample_data']
Success! Here is your data:


,customer_id,signup_date,city_tier,age_group,acquisition_channel,loyalty_tier,preferred_category,skin_type,marketing_consent
0,CUST00001,2024-04-24,Tier 1,18-24,Instagram,Silver,Makeup,Normal,Yes
1,CUST00002,2025-06-01,Tier 2,25-34,Marketplace,Silver,Hair Care,Combination,Yes
2,CUST00003,2025-03-08,Tier 1,25-34,Influencer,NaN,Skin Care,Oily,Yes
3,CUST00004,2025-04-15,Tier 3,25-34,Google Search,NaN,Fragrance,Normal,No
4,CUST00005,2024-08-21,Tier 3,35-44,Organic,Gold,Hair Care,Combination,Yes


In [ ]:
# check for duplicate
print("Duplicate customers:", customers['customer_id'].duplicated().sum())

Duplicate customers: 0


In [ ]:
# Check how many customers have orders
merged = pd.merge(customers, orders, on='customer_id', how='left')
print("Customers with at least one order:", merged['order_id'].notnull().sum())

Customers with at least one order: 10009


**Data Quality Report: Part 1 - Audit**

**​1. Executive Summary**

​The purpose of this audit was to assess the reliability and completeness of the customer data for our upcoming churn prediction project. The dataset has been validated for uniqueness and consistency, confirming it is suitable for further analysis.

**​2. Methodology**

​The data audit was conducted using Python and the pandas library within Google Colab. I merged the customer, order, and support ticket files on the customer_id key to create a comprehensive "Master Table" for analysis.

**​3. Key Findings**

​Data Integrity: I performed a duplication check on the customer_id column and confirmed 0 duplicate records.

​Data Volume: The Master Table contains 10,009 total orders, providing a strong foundation for our modeling efforts.

​Completeness: My analysis of missing values identified that fields such as loyalty_tier and skin_type contain significant missing data, which will need to be addressed before building the prediction model.

**​4. Recommendations**

​Missing Data Handling: I recommend treating missing values in the loyalty_tier and skin_type categories as a distinct "Unknown" category rather than dropping these rows, as this will preserve our sample size.

​Ongoing Validation: Future data imports should be checked against the customer_id master list to ensure there are no orphaned records in the order or support ticket logs.

requirements.txt

pandas>=2.0.0